lms/views/api/canvas_studio.py

```diff
@@ -128,14 +128,16 @@ def list_collection(request):
     permission=Permissions.API,
 )
 def via_url(request):
-    svc = request.find_service(CanvasStudioService)
-    document_url = request.params.get("document_url")
-    media_id = (
-        CanvasStudioService.media_id_from_url(document_url) if document_url else None
+    assignment = request.find_service(name="assignment").get_assignment(
+        request.lti_user.application_instance.tool_consumer_instance_guid,
+        request.lti_user.lti.assignment_id,
```

Comment:
Perhaps it would make sense to add a `get_assignment_for_request` method to `AssignmentService` that handles the lookup of these details? There is a `get_assignment_for_launch` method, but that is specifically for the LTI launch, not other API requests.

Reply:
I reckon it would be very convenient to do: 

request.lti_user.lti.course 
request.lti_user.lti.assignment 

(or rather request.lti.*)

I'll make a note for it :+1: